modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/product_stack.py

```diff
@@ -14,13 +15,115 @@
+        deploy_app_asset: s3_assets.Asset,
+        code_pipeline_deploy_project_name: str,
+        access_token_secret_name: str,
+        repository_owner: str,
+        aws_codeconnection_arn: str,
+    ):
+        # Create GitHub repository
+        github_repo = GitHubRepositoryCreator(
+            self,
+            "DeployAppGitHubRepo",
+            github_token_secret_name=access_token_secret_name,
+            repo_name=f"{sagemaker_project_name}-deploy",
+            repo_description=f"Deployment repository for SageMaker project {sagemaker_project_name}",
+            github_owner=repository_owner,
+            s3_bucket_name=deploy_app_asset.s3_bucket_name,
+            s3_bucket_object_key=deploy_app_asset.s3_object_key,
+            code_connection_arn=aws_codeconnection_arn,
+        )
+        github_env_vars = {
+            "CODE_CONNECTION_ARN": codebuild.BuildEnvironmentVariable(value=aws_codeconnection_arn),
+            "SOURCE_REPOSITORY": codebuild.BuildEnvironmentVariable(
+                value=f"{repository_owner}/{sagemaker_project_name}-deploy"
+            ),
+        }
+        # Append github_env_vars into codebuild_env_vars
+        codebuild_env_vars = {**env_vars, **github_env_vars}
+
+        project = codebuild.Project(
+            self,
+            code_pipeline_deploy_project_name,
+            build_spec=codebuild.BuildSpec.from_object(
+                {
+                    "version": "0.2",
+                    "phases": {
+                        "build": {
+                            "commands": [
+                                "npm install -g aws-cdk",
+                                "python -m pip install -r requirements.txt",
+                                "export REPOSITORY_TYPE=GitHub",
+                                'cdk deploy --require-approval never --app "python app.py" ',
+                            ]
+                        }
+                    },
+                }
+            ),
+            source=codebuild.Source.git_hub(owner=repository_owner, repo=f"{sagemaker_project_name}-deploy"),
+            environment=codebuild.BuildEnvironment(
+                build_image=codebuild.LinuxBuildImage.STANDARD_5_0,
+                environment_variables=codebuild_env_vars,
+            ),
+        )
+        return project, github_repo
+
+    def create_codebuild_with_codecommit(
+        self,
+        env_vars: str,
+        sagemaker_project_name: str,
+        deploy_app_asset: s3_assets.Asset,
+        code_pipeline_deploy_project_name: str,
+    ):
+        # Create CodeCommit repo from seed bucket/key
+        repository = codecommit.Repository(
+            self,
+            "Deploy App Code Repo",
+            repository_name=f"{sagemaker_project_name}-deploy",
+            code=codecommit.Code.from_asset(
+                asset=deploy_app_asset,
+                branch="main",
+            ),
+        )
+        project = codebuild.Project(
```

Comment:
Same here - CodeBuild project in both cases looks the same, the only difference is the source. Unless I am missing anything?

Reply:
Optimised